lib/sensu/api/process.rb

```diff
@@ -794,6 +794,48 @@ def resolve_event(event_json)
           end
         end
       end
+
+      aget %r{^/results?/\*/([\w\.-]+)/?$} do |check_name|
+        keys = "result:*:#{check_name}"
+        settings.redis.keys(keys) do |keys|
+          EM::Iterator.new(keys).map(
```

Comment:
`EM::Iterator` creates anonymous classes, causing memory issues on JRuby. We should probably use a combination of `Proc` and `EM::next_tick`, like: https://github.com/sensu/sensu/blob/master/lib/sensu/server/filter.rb#L262-L272